src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,39 @@
 package core.basesyntax.service;
 
+import static core.basesyntax.db.Storage.people;
+
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LENGTH = 6;
+    private static final int MINIMAL_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user.getLogin() == null || user.getLogin().isEmpty()) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getPassword() == null || user.getPassword().isEmpty()) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null");
+        }
+        if (user.getAge() < MINIMAL_AGE) {
+            throw new RegistrationException("Not valid age: " + user.getAge()
+                    + ". Min allowed age is " + MINIMAL_AGE);
+        }
+
+        if (((people.contains(user.getLogin())
+                && user.getLogin().length() >= MINIMAL_LENGTH)
+                && user.getPassword().length() >= MINIMAL_LENGTH
+                && user.getAge() >= MINIMAL_AGE)) {
```

Comment:
The current logic for checking whether to add a user to storage is incorrect. According to the task requirements, you should ensure that the user's login is unique, and both the login and password should be at least 6 characters long. The current condition incorrectly uses `people.contains(user.getLogin())`, which should instead check if the login is not already present in storage. Also, the conditions should be split into separate checks as per the checklist recommendation: 'Don't create one big if with all conditions. Better to split one big if into several small ones.'